src/OpenLoco/src/World/CompanyAi/CompanyAi.cpp

```diff
@@ -4818,20 +4819,264 @@ namespace OpenLoco
+                        company.var_85C2 = i;
+                        company.var_85C3 &= ~(1U << 0);
+                        company.var_85D0 = thought.stations[i].pos;
+                        company.var_85D4 = thought.stations[i].baseZ;
+                        company.var_85D5 = thought.stations[i].rotation ^ (1U << 1);
+                        return 0;
+                    }
+                    else if (thought.stations[i].var_C & 0b1010)
+                    {
+                        company.var_85C2 = i;
+                        company.var_85C3 |= 1U << 0;
+                        company.var_85D0 = thought.stations[i].pos;
+                        company.var_85D4 = thought.stations[i].baseZ;
+                        company.var_85D5 = thought.stations[i].rotation;
+                        return 0;
+                    }
+                }
+                return 1;
+            }
+            else
+            {
+                // 0x00486D3E
+                const auto pos = World::Pos3(company.var_85D0, company.var_85D4 * World::kSmallZStep);
+                const auto tad = company.var_85D5;
+                const auto queryMods = 0U;
+                const auto requiredMods = 0U;
+                const auto roadObjId = thought.trackObjId & ~(1U << 7);
+                auto* roadObj = ObjectManager::get<RoadObject>(roadObjId);
+                auto compatRoadObjId = roadObj->hasFlags(RoadObjectFlags::unk_03) ? 0xFFU : roadObjId;
+
+                auto [nextPos, nextRotation] = Track::getRoadConnectionEnd(pos, tad);
+                auto rc = Track::getRoadConnectionsAiAllocated(nextPos, nextRotation, company.id(), compatRoadObjId, requiredMods, queryMods);
+                if (rc.connections.empty())
+                {
+                    // 0x00486E63
+                    const auto aiStationIndex = company.var_85C2;
+                    auto& aiStation = thought.stations[aiStationIndex];
+                    if (company.var_85C3 & (1U << 0))
+                    {
+                        if (aiStation.var_C & (1U << 1))
+                        {
+                            aiStation.var_C &= ~(1U << 1);
+                            aiStation.var_C |= (1U << 2);
+                        }
+                        else
+                        {
+                            aiStation.var_C &= ~(1U << 3);
+                            aiStation.var_C |= (1U << 4);
+                        }
+                    }
+                    else
+                    {
+                        if (aiStation.var_B & (1U << 1))
+                        {
+                            aiStation.var_B &= ~(1U << 1);
+                            aiStation.var_B |= (1U << 2);
+                        }
+                        else
+                        {
+                            aiStation.var_B &= ~(1U << 3);
+                            aiStation.var_B |= (1U << 4);
+                        }
+                    }
```

Comment:
Considering the only difference is the variable the flags are applied to... Lambda helper function?